chore/find_javadoc_regression.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+
+import re
+import sys
+from pathlib import Path
+from typing import List, Dict
+
+
+def filter_relevant_lines(lines: List[str]) -> List[str]:
+    """Tries to filter the checkstyle output to only include lines with violations."""
+    new_lines = []
+
+    for line in lines:
+        if re.search(r":\d+:", line) and "ERROR" in line:
+            new_lines.append(line)
+
+    return new_lines
+
+
+def strip_line_number(line: str) -> str:
+    return re.sub(r".java:\d+(:\d+)?:", ".java", line)
+
+
+def strip_line_numbers(lines: List[str]) -> Dict[str, int]:
+    """
+    Strips the violation line numbers to account for shifts.
+    This method returns a dictionary of "stripped line -> Count", as the same
+    error might appear multiple times.
+    """
+    output_dict: Dict[str, int] = dict()
+
+    for line in map(strip_line_number, lines):
+        if line in output_dict:
+            output_dict[line] += 1
+        else:
+            output_dict[line] = 1
+
+    return output_dict
+
+
+def try_match_lines(reference: Dict[str, int], other: Dict[str, int]) -> Dict[str, int]:
+    """Tries to find out which lines are *new* in other and which were already present in the reference."""
+    other = other.copy()
+    for line, amount in reference.items():
+        for _ in range(0, amount):
+            if line in other:
+                if other[line] > 1:
+                    other[line] -= 1
+                else:
+                    other.pop(line)
+
+    # Lines that are only in the reference aren't critical, as they were
+    # apparently fixed in "other"
+    return other
```

Comment:
If you use `collections.Counter`, this can be computed as a subtraction

```suggestion
def try_match_lines(reference: collections.Counter[str], other: collections.Counter[str]) -> Dict[str, int]:
    """Tries to find out which lines are *new* in other and which were already present in the reference."""
    return other - reference
```

Again, maybe it doesn't even need to be a function of its own anymore

Reply:
I think I'll leave it as a function because I like the comment trying to explain why we subtract them in that order. It is probably perfectly clear without it, but I am slightly more positive that I will understand what I did in a year with it present.